src/main.rs

```diff
@@ -552,6 +576,38 @@ fn run() -> Result<(), failure::Error> {
         } else {
             commands::subdomain::get_subdomain(&user, &target)?;
         }
+    } else if let Some(route_matches) = matches.subcommand_matches("route") {
+        let user = settings::global_user::GlobalUser::new()?;
+        let manifest = settings::toml::Manifest::new(config_path)?;
+        let env = matches.value_of("env");
+
+        let env_zone_id = if let Some(environment) = manifest.get_environment(env)? {
+            environment.zone_id.as_ref()
+        } else {
+            None
+        };
+
+        let zone_id: Result<String, failure::Error> = if let Some(zone_id) = env_zone_id {
+            Ok(zone_id.to_string())
+        } else if let Some(zone_id) = manifest.zone_id {
+            Ok(zone_id)
+        } else {
+            failure::bail!(
+                "You must specify a zone_id in `wrangler.toml` to use `wrangler route` commands."
+            )
+        };
+
+        match route_matches.subcommand() {
+            ("list", Some(_)) => {
+                commands::route::list(zone_id?, &user)?;
+            }
+            ("delete", Some(delete_matches)) => {
+                let route_id = delete_matches.value_of("route_id").unwrap();
+                commands::route::delete(zone_id?, &user, route_id)?;
+            }
+            ("", None) => message::warn("route expects a subcommand"),
```

Comment:
technically this should never be reached given that `.setting(AppSettings::SubcommandRequiredElseHelp)` is used above. Think we can delete this line?

Reply:
this is copied from the kv subcommands, which were copied from the clap documentation. atm i'm kinda just going on what i know works, i'm not 100% certain why they specify this.

Reply:
ok! we can keep this for now for consistency, though this is something we can definitely add to a clean up effort when it arrives.